web3.md

```diff
@@ -186,5 +189,21 @@ module WEB3
            <code> CODE </code>
            ...
          </account>
+
+    syntax KItem ::= "#eth_sign"
+ // ----------------------------
+    rule <k> #eth_sign => #signMessage(#privateKey(#parseHexWord(ACCTADDR)),#hashMessage(#unparseByteStack(#parseByteStack(MESSAGE)))) ... </k>
+         <params> [ ACCTADDR, MESSAGE, .JSONList ] </params>
+
+    syntax KItem ::= #signMessage ( String , String )
+ // -------------------------------------------------
+    rule <k> #signMessage(KEY, MHASH) => #sendResponse( "result" : "0x" +String ECDSASign( MHASH, KEY ) ) ... </k>
+
+    syntax String ::= #hashMessage ( String ) [function]
+                    | #privateKey  ( Int )    [function]
+ // ----------------------------------------------------
+    rule #hashMessage( S ) => #unparseByteStack(#parseHexBytes(Keccak256("\x19Ethereum Signed Message:\n" +String Int2String(lengthString(S)) +String S)))
+    rule [[ #privateKey( ADDR ) => #unparseByteStack(#parseByteStack(KEY)) ]]
+         <accountKeys>... ADDR |-> KEY ...</accountKeys>
```

Comment:
I would prefer to inline this directly instead of using a contexual function, so we have a better idea of where it's getting stuck (and so we can send back an error message if the key is not present).

```
    syntax KItem ::= "#eth_sign"
 // ----------------------------
    rule <k> #eth_sign => #signMessage(PRIV_KEY, #hashMessage(...)) ... </k>
         <accountKeys> ... #parseHexWord(ADDR) |-> KEY ... </accountKeys>
    rule <k> #eth_sign => #sendRespones("SOME ERROR MESSAGE") ... </k>

         <accountKeys> ACCTKEYS </accountKeys>
       requires notBool #parseHexWord(ADDR) in_keys(ACCTKEYS)
```